docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -0,0 +1,391 @@
+
+Defining placement of the index in one specific partition:
+
+```sql
+ALTER TABLE table_name ALTER PARTITION partition_name ALTER INDEX index_name
+	{ADD | ALTER} PLACEMENT LABEL=label ROLE=role COUNT=count,
+...
+```
+
+The key format of partitioned index is as `t_{partition_id}_i_{index_id}_r_{pk_value}`. The key range can be inferred by `partition_id` and `index_id`.
+
+When an index is partitioned, defining placement of the whole index at once is not supported. It will involve multiple key ranges, and the scenario of its application is rare.
+
+For example, `t` is a partitioned table and index `idx` is the index on `t`. It’s not supported to do this:
+
+```sql
+ALTER TABLE `t` ALTER INDEX `i` ADD PLACEMENT ...
+```
+
+To alter the placement rule of `idx`, a partition must be specified in the statement.
+
+Currently, global secondary index on partitioned tables is not supported, so it can be ignored for now.
+
+### Sequence placement
+
+Sequence is typically used to allocate ID in INSERT statements, so the placement of sequences affects the latency of INSERT statements.
+
+However, sequence is typically used with cache enabled, which means very few requests are sent to sequence. So defining placement of sequences is not supported for now.
+
+## Drop placement rules
+
+Dropping a placement rule can be achieved by a DROP clause:
+
+```sql
+ALTER TABLE table_name
+	DROP PLACEMENT
+```
+
+Placement rules of indexes and partitions can also be dropped in a similar grammar. The statement must wait until PD returns a message.
+
+## Rule priorities
+
+When several rules are defined for one row, the most granular is chosen for this row. More specifically, the rule priority is: Index > Partition > Table > Database > Default.
+
+For example:
+
+1. At the beginning, all rows are placed based on the default placement rules.
+2. When a placement rule is added on table `t`, all data on `t` is placed based on the rule of `t`.
+3. When a placement rule is added on partition `p0` of `t`, all data on `p0` is placed based on the rule of `p0`, but other partitions stay still.
+4. When the placement rule on `p0` is removed, data on `p0` is placed based on the rule of `t`, just like other partitions.
+
+Rules priorities should be checked once a placement is added, altered, or dropped.
+
+Rule priorities can be implemented by properties `index` and `override`. `override` is alway enabled, and `index` stands for the priority. Specifically, `index` values can be like this:
+
+* `index` of default placement rules is 0
+* `index` of database placement rules is 1
+* `index` of table placement rules is 2
+* `index` of partition placement rules is 3
+* `index` of index placement rules is 4
+
+In such a way, the most granular rule always works.
+
+When multiple placement rules are defined on the same granular object, the new one always overwrites the old one. To achieve this, property `group_id` in placement rules can be used to identify the database object.
```

Comment:
Index and overwrite should only be effective within a group, not across groups. What is the purpose of using group id here?